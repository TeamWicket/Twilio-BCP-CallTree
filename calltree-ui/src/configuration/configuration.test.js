import React from 'react';
import { mount, shallow } from 'enzyme';
import { Admin } from 'react-admin';
import Configuration from '../../../calltree-ui/src/configuration/configuration';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'

describe("Configuration", () => {
    let middlewares;
    let mockStore;
    let initialState;
    let store;
    beforeAll(() => {
        middlewares = [];
        mockStore = configureStore(middlewares);
        initialState = { theme: "light" };
        store = mockStore(initialState);
    })

    it("should match snapshot of Configuration Component", () => {
        expect(shallow(<Admin><Configuration /></Admin>)).toMatchSnapshot();
    })

    it("should check the action being called on button click", () => {
        const wrapper = mount(
            <Provider store={store}>
                <Configuration />
            </Provider >);
        const buttons = wrapper.find("button");
        buttons.at(1).simulate("click");
        expect(store.getActions().pop()).toEqual({ type: 'CHANGE_THEME', payload: 'dark' })
        buttons.at(0).simulate("click");
        expect(store.getActions().pop()).toEqual({ type: 'CHANGE_THEME', payload: 'light' })
    })
})
