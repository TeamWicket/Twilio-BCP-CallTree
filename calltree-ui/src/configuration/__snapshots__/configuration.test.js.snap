// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configuration should match snapshot of Configuration Component 1`] = `ShallowWrapper {}`;
